src/index.ts

```diff
@@ -25,6 +25,24 @@ export interface Collection<K, V> extends Map<K, V> {
 export class Collection<K, V> extends Map<K, V> {
 	public static readonly default: typeof Collection = Collection;
 
+	/**
+	 * Obtains an element if the key exists, otherwise sets and returns the value provided by the default value generator.
+	 *
+	 * @param {*} key - The key to get from if it exists or set otherwise.
+	 * @param {function} defaultValueGenerator - Function that returns a value to be set (if unset) and returned afterwards
```

Comment:
I just realised that we don't use final dots in params, and `function` isn't a valid JSDoc type, `Function` is 😅

Also simplified the description of the second param, since the behaviour is already explained by the method's description.
```suggestion
	 * @param {*} key - The key to get from if it exists or set otherwise
	 * @param {Function} defaultValueGenerator - A function that returns the ensured value
```